contracts/HubPool.sol

```diff
@@ -201,48 +214,101 @@ contract HubPool is Testable, Lockable, MultiCaller, Ownable {
 
     function initiateRelayerRefund(
         uint256[] memory bundleEvaluationBlockNumbers,
-        bytes32 poolRebalanceProof,
-        bytes32 destinationDistributionProof
+        uint64 poolRebalanceLeafCount,
+        bytes32 poolRebalanceRoot,
+        bytes32 destinationDistributionRoot
     ) public {
-        require(currentPendingRefundProposal == false, "Only one proposal at a time");
-        currentPendingRefundProposal = true;
-        relayerRefundRequests.push(
-            RelayerRefundRequest({
-                poolRebalanceProof: poolRebalanceProof,
-                destinationDistributionProof: destinationDistributionProof,
-                status: RefundRequestStatus.Pending
-            })
+        // The most recent refund proposal must be fully claimed before the next relayer refund bundle is initiated.
+        require(
+            relayerRefundRequests.length == 0 || // If this is the first initiated relay refund.
+                relayerRefundRequests[getNumberOfRelayRefunds()].unclaimedPoolRebalanceLeafs == 0,
```

Comment:
well we are not storing the number of relayer refunds, no? we are simply setting this (as the number of leaves) at the beginning. in my mind the big advantage with doing it this way is we know with certainty that there is no pending unclaimed leafs before initialising the next bundle of leafs. this is really critical as you can have a situation with inconsistant state on L1 vs what the bot sees if we do this optimistically (i.e I have not seen the final executed leafs for a set of repayments and therefore dispute the next bundle that comes in as I'm not up too date with master).

obviously, this is not that important if we design the bots well to avoid this but it just seems consistant to store this, with a marginally small gas cost.

put another way, storing this data and having this extra computation is likely less than 10k extra gas per relay refund in total. sending to one L2 costs at least 100k gas so this cost is small when compared to the other costs.

Reply:
That seems reasonable, and I think we should probably just not worry about this until we've nailed down how we're storing everything, as that probably changes a lot of the gas math here.